testnet/stacks-node/src/neon_node.rs

```diff
@@ -2120,13 +2160,52 @@ impl StacksNode {
             "attempt" => attempt
         );
 
-        let res = bitcoin_controller.submit_operation(op, &mut op_signer, attempt);
-        if !res {
-            if !config.node.mock_mining {
-                warn!("Failed to submit Bitcoin transaction");
-                return None;
-            } else {
-                debug!("Mock-mining enabled; not sending Bitcoin transaction");
+        let mut send_tx = true;
+        if cfg!(test) {
```

Comment:
I think this function is already doing a lot for the reader to follow, and they will now have to keep track of a new long cryptic `cfg!(test)` block.

any way to inject from the outside without putting it into the prod code using some kind of "for test" version of an interface?

an alternative would be to make a function in this file called `maybe_test_late_block_commit`.

at the end of the day if that's not worth it and you want to check this in it's fine with me.

Reply:
I agree with this -- this function is already quite complex, and it seems like this is something that would be pretty easy to at split out to another function:

```rust
#[cfg(test)]
fn test_commit_fault_injection(op, op_signer, attempt, bitcoin_controller, block_height) -> bool {
...
}
#[cfg(not(test))]
fn test_commit_fault_injection(op, op_signer, attempt, bitcoin_controller, block_height) -> bool {
   true
}
...
let send_commit = test_commit_fault_injection(op, op_signer, attempt, bitcoin_controller, block_height);
if send_commit {
...
}
...
```